### PR TITLE
fix(payments): only show zip input error onBlur

### LINKED ISSUE #2096: Zip code error jumping the gun
Every time I click into the zip code on the billing form and add a number it immediately gives me an error. Can we remove this and have the error only show up if the user clicks outside of the form element?

Screen Shot 2019-08-06 at 11 21 40 AM

### PR DESCRIPTION
**Pull request by meandavejustice**
- fixes #2096

### CODE REVIEW

File: packages/fxa-payments-server/src/components/PaymentForm/index.tsx [hunk 1]
```diff
@@ -111,7 +116,7 @@ export const PaymentForm = ({
     [validator, onPayment, onPaymentError, stripe]
   );
 
-  stripeElementStyles = checkMedia(matchMedia(SMALL_DEVICE_RULE), stripeElementStyles);
+  stripeElementStyles = checkMedia({ matched: matchMedia(SMALL_DEVICE_RULE), stripeElementStyles });
```
Comment (lmorchard) [inline packages/fxa-payments-server/src/components/PaymentForm/index.tsx]: This looks like a change not related to the zip field

File: packages/fxa-payments-server/src/components/PaymentForm/index.tsx [hunk 2]
```diff
@@ -49,7 +49,11 @@ type StripeElementStyles = {
   lineHeight: string,
 };
 
-export function checkMedia(matched: boolean, stripeElementStyles: { base: StripeElementStyles }) {
+export function checkMedia({ matched, stripeElementStyles }: {
```
Comment (lmorchard) [inline packages/fxa-payments-server/src/components/PaymentForm/index.tsx]: This change looks unrelated too

File: packages/fxa-payments-server/src/components/fields.tsx [hunk 3]
```diff
@@ -130,6 +133,8 @@ export const Input = (props: InputProps) => {
     [name, validator, onValidate, required]
   );
 
+  const onChangeWrapper = (ev: any) => onChange(ev);
```
Comment (lmorchard) [inline packages/fxa-payments-server/src/components/fields.tsx]: Doesn't look like this wrapper does anything? Should also be defined using `React.useCallback` so as not to be re-created on every component render.

File: packages/fxa-payments-server/src/components/PaymentForm/index.tsx (hunk 1 above)
Comment (meandavejustice) [reply packages/fxa-payments-server/src/components/PaymentForm/index.tsx]: This was a typescript checker change

File: packages/fxa-payments-server/src/components/PaymentForm/index.tsx (hunk 2 above)
Comment (meandavejustice) [reply packages/fxa-payments-server/src/components/PaymentForm/index.tsx]: same as ⬆️ (typescript checker)

File: packages/fxa-payments-server/src/components/fields.tsx (hunk 3 above)
Comment (meandavejustice) [reply packages/fxa-payments-server/src/components/fields.tsx]: This is to keep the typescript checker happy. It doesn't like assigning a function with two arguments to `onChange`

**Comment by meandavejustice** [comment]
@lmorchard Thanks for your review, it somehow slipped through my radar. I can pull the stripeElementStyles changes out into a separate pr, add max length, try and figure out the not calling onBlur on selecting checkbox. And the other changes I think we should punt(I can file follow up issues), as I've spent a ton of time on such a minor change already.

**Comment by meandavejustice** [comment]
@lmorchard Pulled out the stripeElementStyles refactoring stuff out and made the `maxlength` updates. Should be pretty clean and straight forward now.

**Comment by lmorchard** [comment]
Yeah, I think this is looking pretty good now. Still see an issue where if the zip code field is valid but not blurred away from, the submit button never enables. But I think that might be an edge case to fix if it ends up being a real problem

**Comment by meandavejustice** [comment]
> Still see an issue where if the zip code field is valid but not blurred away from, the submit button never enables
@lmorchard What are the steps to reproduce? I wanna file a followup issue and I haven't seen that myself

**Comment by lmorchard** [comment]
>What are the steps to reproduce? I wanna file a followup issue and I haven't seen that myself

Basically, this thing [I mentioned earlier](https://github.com/mozilla/fxa/pull/2541#discussion_r328781869):

> If I click the confirmation checkbox first, then fill out all the form fields but never blur away from zip code - the value doesn't get validated and the button never gets enabled. Maybe this is more an edge case? Will probably end up filed as an issue though

**Pull request merged**